src/session_impl.cpp

```diff
@@ -5661,7 +5655,7 @@ namespace aux {
 		m_dht_settings = settings;
 	}
 
-	void session_impl::set_dht_state(dht::dht_state const& state)
+	void session_impl::set_dht_state(dht::dht_state state)
 	{
 		m_dht_state = state;
```

Comment:
that line cost me the read of a dozen articules :) If I understood well, all them seems to indicate that in this particular case, the compiler can infer the "implicit" call to the move assignment operator, but if you ask me, I rather make it explicit.